casper/src/test/scala/coop/rchain/casper/engine/ApproveBlockProtocolTest.scala

```diff
@@ -210,9 +209,10 @@ class ApproveBlockProtocolTest extends FlatSpec with Matchers {
   }
 
   it should "not accept BlockApproved messages signed by not trusted validators" in {
-    implicit val ctx         = TestScheduler()
-    implicit val logStub     = new LogStub[Task]()
-    implicit val metricsTest = new MetricsTestImpl[Task]()
+    implicit val ctx          = TestScheduler()
+    implicit val logStub      = new LogStub[Task]()
+    implicit val eventLogStub = new EventLogStub[Task]()
+    implicit val metricsTest  = new MetricsTestImpl[Task]()
```

Comment:
Why don't we move all those repeated implicit vals to the test's fields?

Reply:
AFAIK the `*Stub`s are used locally to accumulate f.e. logs and verify them in the particular test. ATM I didn't verify the collected events in the `EventLogStub`. A refactoring of Casper unit test is not in the scope of this PR

Reply:
If they need to be local, then they can be re-declared locally. In all ohter cases they should use the same subs, noop-stubs.

> A refactoring of Casper unit test is not in the scope of this PR

Honestly, whenever a change requires changing a dozen of places in one file, I'd say we should require first doing the refactoring that reduces the number of places to one. Especially if it's as simple as in this case.

Even more: you could've declared that new implicit once, globally.

Reply:
Notice I approved the PR, I'm just making a case for a better future :)